### ISSUE TITLE
Deprecation warning for v7 changes links to v5-to-v6 migration guide

### ISSUE THREAD

**Issue by Arkellys** [javierbrea/eslint-plugin-boundaries#477]
I was getting this warning:

```
[boundaries][warning]: [boundaries/dependencies] Detected legacy selector syntax in 3 rule(s) at indices: 0, 1, 2.
  Consider migrating to object-based selectors. More info: https://www.jsboundaries.dev/docs/releases/migration-guides/v5-to-v6/
```

For this conf:

```js
"boundaries/dependencies": ["error", {
    default: "disallow",
    policies: [
      {
        from: { type: "main" },
        allow: [{ to: { type: "core" } }],
      },
      {
        from: { type: "renderers" },
        allow: [
          { to: { type: "renderers", captured: { domain: "common" } } },
          { to: { type: "core" } },
        ],
      },
      {
        from: { type: "preloads" },
        allow: [
          { to: { type: "preloads", captured: { domain: "common" } } },
          { to: { type: "core" } },
        ],
      },
    ],
  },
];
```

The problem here was that I was missing the `{ element: {...} }` wrappers.
The deprecation warning links to the v5-to-v6 migration guide, but this warning is actually related to v7.

I also have to mention that I had no clue what "in 3 rule(s) at indices: 0, 1, 2" even meant. It would be nice to have clearer deprecation warning messages, especially given the number of changes we have to deal with in every major version.

**Comment by javierbrea** [comment]
Hi @Arkellys, thanks for reporting this!

**Regarding the link:** You're absolutely right — the warning is incorrectly pointing to the v5-to-v6 migration guide. I'll get that fixed.

**Regarding the warning message:** The "in 3 rule(s) at indices: 0, 1, 2" part refers to the indices of the `policies` array entries where the legacy syntax was detected, so you can identify which ones need updating. I understand it can be confusing though — the wording should be improved. At the very least, the term "rule(s)" should be "policies", which is the correct terminology since v7. I'm also considering adding more detail about what specific legacy format was detected in each case, to make it more actionable. I'll work on improving this.

**Regarding breaking changes across major versions:** I understand the frustration, and I'm sorry for the inconvenience. The features requested over time made the configuration model increasingly complex, as new capabilities were added incrementally across versions to support restrictions based on any dependency characteristic. The most significant change came between v6 and v7 with the introduction of multi-layer categorization, which allows classifying files independently from multiple axes, not just by the element they belong to — that required a more substantial rethink of the syntax.

I've tried to make these transitions as progressive as possible, avoiding major breaking changes where I could to ease adoption, though I know that hasn't always been easy. That said, I believe the current model is now flexible and powerful enough to cover most architectural scenarios, and much easier to extend without introducing further breaking changes. So I'm hopeful there won't be more deprecations in the near future, and I can focus on adding more value without changing the syntax again.
